runtime/lua/vim/_editor.lua

```diff
@@ -735,6 +735,19 @@ function vim._cs_remote(rcid, server_addr, connect_error, args)
   }
 end
 
+--- Display a deprecation notification to the user.
+---
+---@param name        string     Deprecated function.
+---@param alternative string|nil Preferred alternative function.
+---@param version     string     Neovim version in which the deprecated
```

Comment:
Is this intended as neovim exclusive function? If not, maybe worth removing the `Neovim` here? Plugins could use it as well. (At least right now nothing in the message is neovim specific)

Reply:
Hmm, good point. Thanks.

Reply:
I think it would be helpful to expose it, and add an (optional?) scope argument.

Reply:
Hmm. What would a scope be in this context? Can you give a quick example?

Reply:
neovim, fugitive, gitsigns. The plugin name I believe.

Reply:
@muniter thanks.

I'll give it a shot.

Reply:
Gave it a shot :). Please take a look.

Reply:
I'll provide a counter argument. I don't think exposing a scope argument is worth it, for two reasons: 1) plugins may not want to use the same verbiage that `vim.deprecate` uses, so they will end up writing their own function anyway (we could solve this by adding yet another `message` argument, but then it becomes generalized beyond the point of usefulness), and 2) writing a "deprecated" function is so easy I doubt making it extensible to plugins will even be that useful.

So I'd vote for just keeping this Neovim specific. Just my 2 cents.

Reply:
I was on mobile, so that was a bit brief. My suggestion was to make this 
```
<scope>: <function> is deprecated and will be removed by <date>. <Alternative text if not nil>
```
(Neovim itself may want to use a different verbiage depending on the exact nature of deprecation -- it may not be a 1:1 function replacement.)

In general, the same argument applies to Neovim as well: Is it worth adding a new function to the API for something that's effectively a very thin wrapper over `vim.notify_once`? 

I'd posit what's sauce for the goose is sauce for the gander.

Reply:
> In general, the same argument applies to Neovim as well: Is it worth adding a new function to the API for something that's effectively a very thin wrapper over `vim.notify_once`?

I think so, because I've seen poor wording and missing details in these messages many times over the years. Removing a bit of friction helps us be better about deprecating things and providing notices.



> I don't think exposing a scope argument is worth it, for two reasons: 1) plugins may not want to use the same verbiage that `vim.deprecate` uses (how many plugins use versions anyway?),

That's true but I bet they end up using it :) It's nice to have a standard mechanism that nudges authors towards clear messaging. 

The scope arg can be an optional last one, or we can make this a "kwargs"-style function (probably should be, given the number of params).